### DISCUSSION TITLE
How to open .md document in R Studio Viewer?

### DISCUSSION THREAD

**Discussion by turbanisch** [quarto-dev/quarto-cli#9715] [Q&A]
### Description

I am rendering a Quarto document to GitHub-flavored Markdown (`gfm`). At the end of the rendering process, the result is shown as an HTML preview in the Viewer pane. This preview looks "nice" as it seems to use Quarto's HTML styling. I would like to get a similar preview when I open the generated .md files manually later.

However, in R Studio, all I can do is open the .md file, click on "render" again, only to end up with a preview that does not look as nice (i.e., is not using Quarto's HTML styling). This process also modifies the .md file (for example by adding line breaks) which I do not want.

Is it possible to view .md files with Quarto's HTML styling in the Viewer pane at a later point in time without having to render again?

**Comment by mcanouil** [comment]
I suggest to use the Visual Editor.

Note that a markdown file has nothing to do with HTML/CSS. It's a pure text file.
The "preview" is the doing of the websites such as GitHub, etc. They use their own CSS to display markdown.

**Comment by cderv** [comment]
Quarto has a special feature for `format: gfm` which is a HTML preview mode when `quarto preview` is run. 
See https://quarto.org/docs/output-formats/gfm.html#preview-mode
It is a real preview, in the sense that output for `format: gfm` will be a `.md` file, but a html is rendered with a special github-like HTML format in temp directory, specically for previewing. 

In RStudio, when you click render on a .qmd file, `quarto preview` is ran, and this HTML preview should be opened, the same way as `quarto preview` in terminal. 

So if you have a `.md` or `.qmd` file with `format: gfm` in the header like
```
---
format: gfm
---

# Some content

Here.
````

Then a special github preview will be opened. 

If you have a `.md` extension file and no YAML header in it, then I believe the Preview button in RStudio IDE will call `rmarkdown::render()` to render to HTML

So more generally, what you are asking depend on the content of your file, and the integration made withing the IDE (RStudio IDE and VSCODE will behave differently for example on this). 

If you can  `quarto preview`, then it wil behave on .md like with .qmd 
	- If YAML header with `format`, then it is used
	- Otherwise, default will be to render to HTML 

Hope this helps understand
